eclipse-che/che-server
# Pass status code to Exception on Unauthorised to GitHub Error

### What does this PR do?
Pass status code to Exception on Unauthorised to GitHub Error in order to be able to recognise GitHub Server url if oAuth is not configured and no PAT is present.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-5355
### How to test this PR?

1. Start che from the PR image: `quay.io/eclipse/che-server:pr-618`
2. Start a workspace from a public GitHub server repository.

See: the workspace starts successfully.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
